docs/migration/3_0.md

```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
+- `AuthService.logout` method changed behavior to redirect to `logout` page. Then the method `AuthService.coreLogout` will be dispatched and perform operations previously done by `logout` method (Logout action dispatch, clearing local state, revoking tokens).
+- `AuthService.getClientToken`, `AuthService.refreshClientToken` and `AuthService.isClientTokenLoaded` were moved to `ClientTokenService`.
+
+#### Config
+
+- `AuthConfig` no longer extends `OccConfig`.
+- `login` and `revoke` endpoints were removed from `OccConfig`. `login` endpoint is now available under `tokenEndpoint` property in `AuthConfig`. `revoke` endpoint is available under `revokeEndpoint` property in `AuthConfig`.
+- `storageSync` configuration for `auth` branch in ngrx store was removed. State of token, userId is now synchronized with `AuthStatePersistenceService`. Override this service if you want to sync more properties to localStorage (eg. `refresh_token`).
+
+### Kyma module
+
+Kyma module were removed with all it's code. We expose the same functionality through configuration of auth.
+
+To fetch OpenId token along with access token in Resource Owner Password Flow you have to use following configuration.
+
+``` ts
+authentication: {
+  client_id: 'client4kyma',
+  client_secret: 'secret',
+  OAuthLibConfig: {
+    responseType: 'id_token',
+    scope: 'openid',
+    customTokenParameters: ['token_type', 'id_token'],
+  },
+},
+```
+
+Then you can access OpenId token with `OAuthLibWrapperService.getIdToken` method. For more options related to OpenId token look into `angular-oauth2-oidc` library documentation.
+
+### ASM module refactor
+
+- `getCustomerSupportAgentTokenState`, `getCustomerSupportAgentToken` and `getCustomerSupportAgentTokenLoading` were removed from `AsmSelectors`. To get token use `AuthStorageService.getToken` and `AsmAuthStorageService.getTokenTarget` to check if it belongs to CS agent.
+- Effects in `AsmModule` now instead of `makeErrorSerializable` use `normalizeHttpError` for error transformation.
+- `storageSync` configuration for `asm` branch in ngrx store was removed. State of ASM UI, tokens is now synchronized with `AsmStatePersistenceService`.
+- `CSAGENT_TOKEN_DATA` variable was removed.
+- `AsmState.csagentToken` was removed. Token is now stored in `AuthStorageService`. Check `AsmAuthStorageService.getTokenTarget` to validate if the token belongs to the CS agent.
+- `AsmActions` no longer contains actions related to customer agent token. Instead interact directly with `CsAgentAuthService`.
+- `CustomerSupportAgentTokenInterceptor` interceptor was removed. Token and error handling for CS agent requests are now handled by `AuthInterceptor` and `AsmAuthHttpHeaderService`.
+
+#### AsmAuthService
+
+Service was renamed to `CsAgentAuthService`. `AsmAuthService` is now responsible for making `AuthService` aware of ASM and adjusts it for CS agent support.
```

Comment:
What about adding removed-api-migration for `AsmAuthService` with comment:
The `AsmAuthService` was renamed to `CsAgentAuthService`. That said, the new service `AsmAuthService` was introduced in 3.0, which has a different purpose - it's responsible for making `AuthService` aware of ASM and adjusts it for CS agent support.